lib/diagnostic.js

```diff
@@ -1,21 +1,38 @@
 'use strict'
 
 // Reuse functions wherever possible.
-
+// create a string
+const paragraph = 'Hello Hello my name is Nikki'
 // 1. Write a function that takes a string argument
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+
+// split the words
+const words = paragraph.split(' ')
+// words => ["Hello", "Hello", my", "name", "is", "Nikki"]
+
+const normalizeWords = function () {
+	for (let i = 0; i < words.length; i++) {
+	  normalizedWords[i] = words[i].toUpperCase()
+	}
 }
+// normalizedWords => ["HELLO", "HELLO", "MY", "NAME", "IS", "NIKKI"]
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWordsAsKeys = function (/* */) {
+	for (let i = 0; i < normalizedWords.length; i++) {
+	  	if (uniqueWordsAsKeys[normalizedWords[i]] === undefined) {
```

Comment:
`uniqueWordsAsKeys` isn't an object, so accessing properties on it in this way won't work. I think you probably want to make it and object `{}`